src/tests/system/tests/test_gpo.py

```diff
@@ -0,0 +1,1052 @@
+           SeInteractiveLogonRight key. Add 'deny_user1 to SeDenyInteractiveLogonRight key
+        3. Create and link the GPO 'unreadable policy', and set the permissions for
+           'Authenticated Users' to 'None'
+        4. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing' and
+           'ad_gpo_ignore_unreadable' = 'True
+        5. Start SSSD
+    :steps:
+        1. Authenticate as 'user1'
+        2. Authenticate as 'deny_user1'
+    :expectedresults:
+        1. 'user1' authentication is successful
+        2. 'deny_user1' authentication is unsuccessful
+    :customerscenario: True
+    """
+    user1 = ad.user("user1").add()
+    deny_user1 = ad.user("deny_user1").add()
+
+    ad.gpo("site policy").add().policy(
+        {
+            "SeInteractiveLogonRight": [user1, ad.group("Domain Admins")],
+            "SeDenyInteractiveLogonRight": [deny_user1],
+        }
+    ).link()
+
+    ad.gpo("unreadable policy").add().link().permissions("Authenticated Users", "None", "Group")
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.domain["ad_gpo_ignore_unreadable"] = "True"
+    client.sssd.start()
+
+    assert client.auth.parametrize(method).password("user1", password="Secret123")
+    assert not client.auth.parametrize(method).password("deny_user1", password="Secret123")
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["ssh", "su"])
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.ticket(bz=2151450)
+def test_gpo__works_when_auto_private_groups_is_set_true(client: Client, ad: AD, method: str):
+    """
+    :title: Group policy object host based access control works when auto_private_groups is set to true.
+    :description:
+        This tests for a bug where the primary group is not returned when the user is looked up.
+    :setup:
+        1. Create the following user 'user1'
+        2. Create the following group 'group' and add 'user1' to the group
+        3. Create and link the GPO 'site policy' and add 'user1', groups, 'group' and 'Domain Admins' to
+           SeInteractiveLogonRight key.
+        4. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing' and 'ldap_use_tokengroup' = 'False'
+        5. Start SSSD
+    :steps:
+        1. Authenticate as 'user1'
+        2. Id as 'user1'
+    :expectedresults:
+        1. 'user1' authentication is successful
+        2. Primary group 'Domain Users' is listed
+    :customerscenario: True
+    """
+    user1 = ad.user("user1").add()
+    group = ad.group("group").add().add_members([user1])
+
+    ad.gpo("site policy").add().policy(
+        {
+            "SeInteractiveLogonRight": [user1, group, ad.group("Domain Admins")],
+            "SeDenyInteractiveLogonRight": [],
+        }
+    ).link()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.domain["auto_private_groups"] = "true"
+    client.sssd.start()
+
```

Comment:
```suggestion
user1_id = client.tools.id("user1")
```

Reply:
The static analysis is not happy about accessing memberof on id tham might be None

Reply:
I'm fine with it being unhappy. ;)